samcli/local/apigw/service_error_responses.py

```diff
@@ -51,7 +52,7 @@ def lambda_failure_response(*args):
         :return: A Flask Response
         """
         response_data = jsonify(ServiceErrorResponses._LAMBDA_FAILURE)
-        return make_response(response_data, ServiceErrorResponses.HTTP_STATUS_CODE_502)
+        return make_response(response_data, ServiceErrorResponses.HTTP_STATUS_CODE_500)
```

Comment:
How did we validate this? The issue states the current code returns 200, so why are we adjusting the 502 to a 500 error?

Reply:
I changed this part because the lambda_failure message attached to this error is Internal Server Error, however when I look at http codes for 502 it's bad gateway which I interpreted as there was an issue here with the service code/message not coordinating properly. Let me know what you think

Reply:
The change I made before updated it so lambda internal error doesn't return 200 it throws a internal server error, however I saw even after throwing it it returned a 502 error which is separate from what AWS throws which is 500, I saw this discrepancy in the error codes, which i labeled above.